types/node/index.d.ts

```diff
@@ -772,23 +773,29 @@ declare module "http" {
     import * as stream from "stream";
     import { URL } from "url";
 
-    export interface RequestOptions {
+    export interface ClientRequestArgs {
         protocol?: string;
         host?: string;
         hostname?: string;
         family?: number;
         port?: number | string;
+        defaultPort?: number | string;
         localAddress?: string;
         socketPath?: string;
         method?: string;
         path?: string;
         headers?: { [key: string]: any };
         auth?: string;
         agent?: Agent | boolean;
+        _defaultAgent?: Agent,
         timeout?: number;
+        rejectUnauthorized?: boolean;
```

Comment:
https://github.com/nodejs/node/blob/master/lib/_http_client.js#L251

`rejectUnauthorized` is used here. it will be passed to underlying agent if socketpath provided